hepcrawl/spiders/aps_spider.py

```diff
@@ -41,43 +42,47 @@ class APSSpider(StatefulSpider):
     Example:
         Using the APS spider::
 
-            $ scrapy crawl APS -a 'from_date=2016-05-01' -a 'until_date=2016-05-15' -a 'set=openaccess'
+            $ scrapy crawl APS -a 'from_date=2016-05-01' -a 'until_date=2016-05-15' -a 'sets=openaccess'
 
     .. _See documentation here:
         http://harvest.aps.org/docs/harvest-api#endpoints
+
+    Note:
+        Selecting specific journals is not supported for technical reasons as it's incompatible with the way the last run time is stored.
     """
     name = 'APS'
     aps_base_url = "http://harvest.aps.org/v2/journals/articles"
 
     @strict_kwargs
     def __init__(self, url=None, from_date=None, until_date=None,
-                 date="published", journals=None, sets=None, per_page=100,
+                 date="published", sets=None, per_page=100,
                  **kwargs):
         """Construct APS spider."""
         super(APSSpider, self).__init__(**kwargs)
-        if url is None:
-            # We Construct.
-            params = {}
-            if from_date:
-                params['from'] = from_date
-            if until_date:
-                params['until'] = until_date
-            if date:
-                params['date'] = date
-            if journals:
-                params['journals'] = journals
-            if per_page:
-                params['per_page'] = per_page
-            if sets:
-                params['set'] = sets
-
-            # Put it together: furl is awesome
-            url = furl(APSSpider.aps_base_url).add(params).url
-        self.url = url
+        self._set = sets
+        self.from_date = from_date
+        self.until_date = until_date
+        self.per_page = per_page
+        self.date = date
+
+
+    @property
+    def url(self):
+        params = {
+            'sets': self.set,
+            'from_date': self.from_date or self.resume_from(set_=self._set),
+            'until_date': self.until_date,
+            'per_page': self.per_page,
+            'date': self.date
+        }
+        return furl(APSSpider.aps_base_url).add(params).url
```

Comment:
are the `None` checks that were done previously useless?

Reply:
I tried it out and it does matter, so changing it back to the old way with the checks